openlibrary/plugins/openlibrary/lists.py

```diff
@@ -152,6 +219,70 @@ def render(self, doc, lists):
         return render_template("lists/lists.html", doc, lists)
 
 
+class lists_edit(delegate.page):
+    path = r"(/people/[^/]+)(/lists/OL\d+L)/edit"
+
+    def GET(self, user_key: str, list_key: str):  # type: ignore[override]
+        key = user_key + list_key
+        if not web.ctx.site.can_write(key):
+            return render_template(
+                "permission_denied",
+                web.ctx.fullpath,
+                f"Permission denied to edit {key}.",
+            )
+
+        lst = web.ctx.site.get(key)
+        if lst is None:
+            raise web.notfound()
+        return render_template("type/list/edit", lst, edit=True)
+
+    def POST(self, user_key: str, list_key: str | None = None):  # type: ignore[override]
+        key = user_key
+        if list_key:
+            key += list_key
+
+        if not web.ctx.site.can_write(key):
```

Comment:
This lets admins edit other people's lists (we instead may want to add a user key-check to prevent this)

Reply:
if user_key == current_user.key

Reply:
Decided to hide the edit button instead to avoid accidental editing.